frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/MergeParentIntoCurrentBranchAction.java

```diff
@@ -0,0 +1,35 @@
+package com.virtuslab.gitmachete.frontend.actions.toolbar;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import io.vavr.control.Option;
+import lombok.val;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+
+import com.virtuslab.gitmachete.backend.api.SyncToParentStatus;
+import com.virtuslab.gitmachete.frontend.actions.base.BaseMergeParentIntoBranchAction;
+
+public class MergeParentIntoCurrentBranchAction extends BaseMergeParentIntoBranchAction {
+  @Override
+  public Option<String> getNameOfBranchUnderAction(AnActionEvent anActionEvent) {
+    return getCurrentBranchNameIfManaged(anActionEvent);
+  }
+
+  @Override
+  @UIEffect
+  protected void onUpdate(AnActionEvent anActionEvent) {
+    super.onUpdate(anActionEvent);
+
+    val presentation = anActionEvent.getPresentation();
+    if (!presentation.isVisible()) {
+      return;
+    }
+
+    val isInSyncToParent = getCurrentBranchNameIfManaged(anActionEvent)
+        .flatMap(bn -> getManagedBranchByName(anActionEvent, bn))
+        .flatMap(b -> b.isNonRoot() ? Option.some(b.asNonRoot()) : Option.none())
+        .map(nrb -> nrb.getSyncToParentStatus() == SyncToParentStatus.InSync)
+        .getOrElse(false);
+
+    presentation.setVisible(isInSyncToParent);
```

Comment:
Hmmm so the action is only visible when the branch is **in sync** with parent? 🤔 it should rather be the reverse, merging parent (e.g. `develop`) into branch is typically performed when the branch is **out of sync** with parent. I'd expect the merge action to only be displayed when the edge is **red**